### PR TITLE
Added function for request recreation that considers the writeable re…

### PR DESCRIPTION
**Pull request by stevanbz**
…gistry used for parsing the aggregations

Added unit tests that checks if the bucket level monitor requests are created and ser/de correctly

Signed-off-by: Stevan Buzejic <stevan.buzejic@htecgroup.com>

### Description
This PR enables serialization/de-serialization of bucket level monitors. 
Adds two new methods - recreateObject which is responsible to re-create objects by using the namedWriteableRegistry
and indexMonitor - which calls recreateObject taking into the account namedWriteableRegistry
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/commons/alerting/AlertingPluginInterface.kt [hunk 1]
```diff
@@ -50,6 +51,31 @@ object AlertingPluginInterface {
             }
         )
     }
+    /**
```
Comment (eirsep) [inline src/main/kotlin/org/opensearch/commons/alerting/AlertingPluginInterface.kt:54]: let's not overload index monitor method. Let's change the original method's signature

File: src/test/kotlin/org/opensearch/commons/alerting/action/IndexMonitorRequestTests.kt [hunk 2]
```diff
@@ -32,6 +38,42 @@ class IndexMonitorRequestTests {
...
+    }
+
+    @Test
+    fun `Index bucket monitor serialize and deserialize transport object should be equal`() {
+        val req = IndexMonitorRequest(
+            "1234", 1L, 2L, WriteRequest.RefreshPolicy.IMMEDIATE, RestRequest.Method.POST,
+            randomBucketLevelMonitor()
+        )
+
+        val recreatedObject = recreateObject(req, NamedWriteableRegistry(SearchModule(Settings.EMPTY, emptyList()).namedWriteables)) { IndexMonitorRequest(it) }
```
Comment (eirsep) [inline src/test/kotlin/org/opensearch/commons/alerting/action/IndexMonitorRequestTests.kt:68]: let's have a test method where recreate object method WITHOUT NamedWriteableRegistry fails  and in the same method let's verify that recreate Object WITH NamedWriteableRegistry passes using an actual aggregation in IndexMonitorRequest.

**Review by eirsep — CHANGES_REQUESTED** [review]
Thanks for catching this issue and fixing it Stevan. Left a couple minor comments

File: src/test/kotlin/org/opensearch/commons/alerting/action/IndexMonitorRequestTests.kt (hunk 2 above)
Comment (stevanbz) [reply src/test/kotlin/org/opensearch/commons/alerting/action/IndexMonitorRequestTests.kt:68]: Good point. Let me try to do that

File: src/main/kotlin/org/opensearch/commons/alerting/AlertingPluginInterface.kt (hunk 1 above)
Comment (stevanbz) [reply src/main/kotlin/org/opensearch/commons/alerting/AlertingPluginInterface.kt:54]: Yeah sure. Tnx

**Review by eirsep — APPROVED** [review]

**Review by getsaurabh02 — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**